### PR TITLE
[Refactor] Refactor ShowExecutor to Visitor pattern to facilitate future expansion

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -268,2642 +269,2507 @@ public class ShowExecutor {
...
-            handleShowDictionary();
-        } else if (stmt instanceof ShowBackendBlackListStmt) {
-            handleBackendBlackList();
-        } else {
-            handleEmpty();
-        }
-
-        List<List<String>> rows = doPredicate(stmt, stmt.getMetaData(), resultSet.getResultRows());
-        return new ShowResultSet(resultSet.getMetaData(), rows);
+    public ShowResultSet execute(ShowStmt statement, ConnectContext context) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:275]: looks like this method is good to be `static`, so there is no need `new ShowExecutor().execute()`, just `ShowExecutor.execute()`

File: fe/fe-core/src/main/java/com/starrocks/analysis/FunctionalExprProvider.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import com.starrocks.common.AnalysisException;
 import com.starrocks.common.util.TimeUtils;
 import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.analyzer.SemanticException;
```
[the remark this answers is not in the document]
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/analysis/FunctionalExprProvider.java:25]: There are tons of `catch (AnalysisException e)` there in codebase, this exception replacement might cause all the catch branch changes, and hence different error handling and potentially user behavior change.

If we want to completely eliminating the  `AnalysisException` exception, suggest change the codebase all over together.

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:275]: Static ones are not easy to dependency injection. Subsequent PRs will use ShowExecutor as a globally unique object.

File: fe/fe-core/src/main/java/com/starrocks/analysis/FunctionalExprProvider.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/analysis/FunctionalExprProvider.java:25]: There are too many, and it is impossible to modify them at once. It has been checked here that there will be no logical impact.

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Pull request merged**